### PR TITLE
Add reset! function for KrylovStats

### PR DESCRIPTION
**Pull request by amontoison**
@tmigot

### CODE REVIEW

File: src/krylov_stats.jl [hunk 1]
```diff
@@ -10,11 +10,16 @@ Type for statistics returned by the majority of Krylov solvers, the attributes a
...
   inconsistent :: Bool
-  residuals :: Vector{T}
-  Aresiduals :: Vector{T}
-  status :: String
+  residuals    :: Vector{T}
+  Aresiduals   :: Vector{T}
+  status       :: String
+end
+
+function reset!(stats :: SimpleStats)
```
Comment (tmigot) [inline src/krylov_stats.jl:20]: I lke the function `reset!` overall. What about we try to make something generic like the `show` function?
For instance [not tested]:
```
for f in ["Simple", "Lanczos", "Symmlq", "Adjoint", "LNLQ", "LSLQ"]
  T = Meta.parse("Krylov." * f * "Stats{S}")
  @eval function reset!(stats :: $T) where S
    for field in fieldnames($T)
      statfield = getfield(stats, field)
      if isa(statfield, AbstractVector)
        empty!(statfield)
      end
    end
  end
end
```

File: test/test_stats.jl [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-  storage_type = typeof(solver)
+  storage_type = typeof(stats)
   expected = """Simple stats
   solved: true
   inconsistent: true
   residuals: []
   Aresiduals: []
   status: t"""
   @test strip.(split(chomp(showed), "\n")) == strip.(split(chomp(expected), "\n"))
+  Krylov.reset!(stats)
```
Comment (tmigot) [inline test/test_stats.jl:14]: This seems ok for the coverage. We should probably test it after a solver call, and check whether it worked?

Same for the others `Krylov.reset!(stats)` below.

File: src/krylov_stats.jl (hunk 1 above)
Comment (amontoison) [reply src/krylov_stats.jl:20]: The issue is the call to `fieldnames($T)` that allocates a vector of symbols.
It's for that reason that I did a `reset!` function for each KrylovStats.

Comment (tmigot) [reply src/krylov_stats.jl:20]: Oh... I checked and we might use `for i=1:5 fieldname(SimpleStats{Float64}, i) end` maybe.

**Comment by amontoison** [comment]
You can rebase and merge @tmigot if you want to work on #361 tomorrow.

File: src/krylov_stats.jl (hunk 1 above)
Comment (tmigot) [reply src/krylov_stats.jl]: Last comment :). Wouldn't it be better to have the functions `reset!` and `show` in the same loop `for f in ["Simple", "Lanczos", "Symmlq", "Adjoint", "LNLQ", "LSLQ"]` ?

**Review by tmigot — APPROVED** [review]

**Pull request merged**